### PR TITLE
Core/utils.go unit tests

### PR DESCRIPTION
**Pull request by krzysztof-jastrzebski**

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/test/test_cloud_provider.go [hunk 1]
```diff
@@ -318,6 +318,9 @@ func (tng *TestNodeGroup) Autoprovisioned() bool {
 
 // TemplateNodeInfo returns a node template for this node group.
 func (tng *TestNodeGroup) TemplateNodeInfo() (*schedulercache.NodeInfo, error) {
+	tng.Lock()
```
Comment (mwielgus) [inline cluster-autoscaler/cloudprovider/test/test_cloud_provider.go]: everything here is constant from the creation, no need for locking.

**Comment by mwielgus** [comment]
How much does it improve the coverage?

**Comment by mwielgus** [comment]
Verify fails
```
!!! 'gofmt -s' needs to be run on the following files: 
./cluster-autoscaler/core/utils_test.go
```

File: cluster-autoscaler/cloudprovider/test/test_cloud_provider.go [hunk 2]
```diff
@@ -330,7 +333,7 @@ func (tng *TestNodeGroup) TemplateNodeInfo() (*schedulercache.NodeInfo, error) {
 	}
 	template, found := tng.cloudProvider.machineTemplates[tng.id]
 	if !found {
-		return nil, fmt.Errorf("No template declared for %s", tng.id)
+		return nil, cloudprovider.ErrNotImplemented
```
[the remark this answers is not in the document]
Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/cloudprovider/test/test_cloud_provider.go]: I made this change because I want to be able to use NodeInfo without TemplateNodeInfo and NodeInfo with TemplateNodeInfo in one unit test. In previous version if I define machine template for one NodeInfo then I need to define it for all NodeInfos. Is returning an error so crucial? This is an util class for testing.

**Comment by krzysztof-jastrzebski** [comment]
Coverage improvement:
package cluster-autoscaler.core - 58,2% -> 64,5%
file core/utils.go - 63,8% - >86,2%

File: cluster-autoscaler/cloudprovider/test/test_cloud_provider.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mwielgus) [reply cluster-autoscaler/cloudprovider/test/test_cloud_provider.go]: This error is missleading. The implementation for this method is provided. The only thing is that the author of the test didn't provide the template for one of the node groups.

[the remark this answers is not in the document]
Comment (aleksandra-malinowska) [reply cluster-autoscaler/cloudprovider/test/test_cloud_provider.go]: The edge case in test appears to be a nodegroup with only unready nodes. In this case, GCE cloud provider attempts to retrieve instance template from MIG, and any error generated at this point comes directly from cloud provider. Not adding template to test cloud provider simulates such error. I think changing this would reduce the mocks usefulness by taking away the ability to test this scenario.

But I agree that failing at this point is not necessarily the desired behavior, as a single unhealthy node group could block scale up. I think fixing it would require change to GetNodeInfosForGroups, as indicated here: 
https://github.com/kubernetes/autoscaler/blob/f9cabf3a1a648af2caaaec87dea781446f15f13c/cluster-autoscaler/core/utils.go#L158
